packages/excalidraw/element/binding.ts

```diff
@@ -2233,13 +2233,12 @@ export const getArrowLocalFixedPoints = (
 export const normalizeFixedPoint = <T extends FixedPoint | null>(
   fixedPoint: T,
 ): T extends null ? null : FixedPoint => {
-  return (
-    fixedPoint
-      ? fixedPoint.map((ratio) =>
-          // Do not allow a precise 0.5 for fixed point ratio
-          // to avoid jumping arrow heading due to floating point imprecision
-          ratio === 0.5 ? 0.5001 : ratio,
-        )
-      : null
-  ) as T extends null ? null : FixedPoint;
+  // Do not allow a precise 0.5 for fixed point ratio
+  // to avoid jumping arrow heading due to floating point imprecision
+  if (fixedPoint && (fixedPoint[0] === 0.5 || fixedPoint[1] === 0.5)) {
+    return fixedPoint.map((ratio) =>
```

Comment:
The map could also be removed for a duplication if it's critical.

Reply:
do you mean from perf perspective? I was checking and the main issue would be mapping even if necessary. If done once it's fine.